File: api/src/main/java/io/github/jafarlihi/symposium/model/Thread.java
package io.github.jafarlihi.symposium.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "threads")
@Data
public class Thread {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long userId;
    private String title;
    private Long categoryId;
    private LocalDateTime createdAt;
}

File: api/models/thread.go
package models

import (
	"time"
)

type Thread struct {
	ID         uint32    `json:"id"`
	UserID     uint32    `json:"userID"`
	Title      string    `json:"title"`
	PostCount  uint32    `json:"postCount"`
	CategoryID uint32    `json:"categoryID"`
	CreatedAt  time.Time `json:"createdAt"`
}
